sqlglot/dialects/risingwave.py

```diff
@@ -1,6 +1,501 @@
+            *DB_CREATABLES,
+        }
+
+        ID_VAR_TOKENS = {
+            *Postgres.Parser.ID_VAR_TOKENS,
+            *CREATABLES,
+        }
+
+        def _parse_table_hints(self) -> t.Optional[t.List[exp.Expression]]:
+            # There is no hint in risingwave.
+            # Do nothing here to avoid WITH keywords conflict in CREATE SINK statement.
+            return None
+
+        def _parse_include(self) -> t.Optional[exp.Expression]:
+            column_type = self._parse_var_or_string()
+            column_alias = None
+            inner_field = None
+            header_inner_expect_type = None
+            if not column_type:
+                return None
+            if not self._match(TokenType.ALIAS, advance=False) and not self._match_texts(
+                "INCLUDE", advance=False
+            ):
+                inner_field = self._parse_var_or_string()
+                if not self._match(TokenType.ALIAS, advance=False) and not self._match_texts(
+                    "INCLUDE", advance=False
+                ):
+                    header_inner_expect_type = self._parse_var_or_string()
+
+            if self._match(TokenType.ALIAS):
+                column_alias = self._parse_var_or_string()
+            include_property = self.expression(exp.IncludeProperty, column_type=column_type)
+            if column_alias:
+                include_property.set("column_alias", column_alias)
+            if inner_field:
+                include_property.set("inner_field", inner_field)
+            if header_inner_expect_type:
+                include_property.set("header_inner_expect_type", header_inner_expect_type)
+
+            return include_property
+
+        def _parse_encode_property(self, is_key: bool) -> t.Optional[exp.Expression]:
+            encode_var = self._parse_var_or_string()
+            encode_property: t.Optional[exp.Expression] = self.expression(
+                exp.EncodeProperty, this=encode_var, is_key=is_key
+            )
+            # encode_property: t.Optional[exp.Expression] = self._parse_property_assignment(exp.EncodeProperty)
+            expressions = None
+            if not encode_property:
+                return None
+            if self._match(TokenType.L_PAREN, advance=False):
+                expressions = exp.Properties(expressions=self._parse_wrapped_properties())
+            if expressions:
+                encode_property.set("expressions", expressions)
+            return encode_property
+
+        def _parse_property(self) -> t.Optional[exp.Expression]:
+            # Parse risingwave specific properties.
+            if self._match_texts("ENCODE"):
+                return self._parse_encode_property(is_key=False)
+
+            if self._match_text_seq("KEY", "ENCODE"):
+                return self._parse_encode_property(is_key=True)
+
+            if self._match_texts("INCLUDE"):
+                return self._parse_include()
+
+            return super()._parse_property()
```

Comment:
Can we get rid of this method override and use `PROPERTY_PARSERS` instead?

Reply:
It would work for `ENCODE` and `INCLUDE`, but since `KEY ENCODE` is a text seq, it cannot be matched via `_match_texts`. May I ask is there some good solution?

Reply:
You can add `"KEY"` and then try to match `"ENCODE"` -- if you manage to do it, then you can use the encode parser. Otherwise, you'd retreat to the previous token to unconsume `KEY` and return `None`.